os/include/tinyara/sched.h

```diff
@@ -631,6 +631,9 @@ struct tcb_s {
 	uint32_t ram_size;		/* Size of RAM partition for this app */
 	uint32_t uspace;		/* User space object for app binary */
 	uint32_t uheap;			/* User heap object pointer */
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+	uint8_t app_id;			/* Application id to be used to index into the app heap table in common binary */
```

Comment:
We already have `binary_idx` in task_group binary_s. How about using this?

Reply:
@jeongchanKim 
I had made the changes to use the binary_idx from tg_bininfo object. However, the tg_bininfo object is not propagated to all the tasks. When I tried to copy the tg_bininfo to the newly created tasks, then there is some issue and kernel_tc is getting stuck at various points. I am not sure why this issue is happening. 

However, i think it is better to have the app id variable directly in the tcb_s structure so that we can access it easily during a context switch. If we use tg_bininfo or tg_binid for this purpose, then getting the binary_idx would require more indirections, or calling an api to get binary_idx from tg_binid. Both these will be more time consuming and better to avoid during context switch.. So, i have reverted my previous changes and maintained the app_id variable.

Reply:
"i think it is better to have the app id variable directly in the tcb_s structure so that we can access it easily during a context switch"
I would agree to keep app_id in tcb_s structure. it would be more straight forward.